tools/downloader/networks_compiler.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import sys
+import common
+from pathlib import Path
+
+def compile(reporter, compiler_path, model, model_precision, args, output_dir):
+    (output_dir / model.subdirectory).mkdir(parents=True, exist_ok=True)
+
+    compile_cmd = [str(compiler_path),
+        '-m={}'.format(args.model_dir / model.subdirectory / model_precision / (model.name + '.xml')),
+        '-d={}'.format(args.target_device),
+        '-ip={}'.format('U8' if args.input_precision is None else args.input_precision),
+        '-op={}'.format('FP32' if args.output_precision is None else args.output_precision),
+        '-o={}'.format(output_dir / model.subdirectory / model_precision /(model.name + '.blob')),
+    ]
+    reporter.print_section_heading('{}Compiling {} to BLOB ({})',
+        '(DRY RUN) ' if args.dry_run else '', model.name, model_precision)
+
+    reporter.print('Conversion command: {}', common.command_string(compile_cmd))
+    if not args.dry_run:
+        reporter.print(flush=True)
+        if not reporter.job_context.subprocess(compile_cmd):
+            return False
+    reporter.print()
+    return True
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-d', '--model_dir', type=Path, metavar='DIR',
+        default=Path.cwd(), help='root of the directory tree with IR model files')
+    parser.add_argument('-o', '--output_dir', type=Path, metavar='DIR',
+        help='root of the directory tree to place compiled models files into')
+    parser.add_argument('--name', metavar='PAT[,PAT...]',
+        help='compile only models whose names match at least one of the specified patterns')
+    parser.add_argument('--list', type=Path, metavar='FILE.LST',
+        help='compile only models whose names match at least one of the patterns in the specified file')
+    parser.add_argument('-ip', '--input_precision', dest='input_precision',
+        help='Input precision of compiled network')
+    parser.add_argument('-op', '--output_precision', dest='output_precision',
+        help='output_precision of compiled network')
```

Comment:
Short options should not consist of more than one letter (and I don't think short options are needed here at all).